tests/unit/test_business_logic.py

```diff
@@ -1441,6 +1442,201 @@ def test_add_file_to_request_with_filetype(bll, filetype, success):
             bll.add_file_to_request(release_request, path, author, filetype=filetype)
 
 
+def test_update_file_in_request_invalid_file_type(bll):
+    author = factories.create_user("author", ["workspace"], False)
+
+    relpath = UrlPath("path/file.foo")
+    workspace = factories.create_workspace("workspace")
+    factories.write_workspace_file(workspace, relpath)
+    release_request = factories.create_release_request(
+        "workspace",
+        user=author,
+    )
+
+    # This code reproduces bll.add_file_to_request() - except without the
+    # safeguards - in order that we can give update_file_in_request()
+    # the correct context to hit this test
```

Comment:
There's another test that creates a request file with an invalid filetype so it can test that you can't release files of type .foo. It does it by patching the LEVEL4_FILE_TYPES so it can make the request. 
i.e. I think you should be able to do this, and avoid the lengthy workaround:
```
with patch("airlock.utils.LEVEL4_FILE_TYPES", [".foo"]):
    release_request = factories.create_release_request_at_status(
        "workspace",
        author=author,
        files=[factories.request_file(path="path/file.foo"]
    )
```

Reply:
great shout, done